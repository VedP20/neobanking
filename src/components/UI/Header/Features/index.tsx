import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';

const FeatureItem = ({iconName, text}: any) => {
  return (
    <View style={styles.featureContainer}>
      <TouchableOpacity style={styles.iconContainer}>
        <Icon name={iconName} color={'gray'} size={24} />
      </TouchableOpacity>
      <Text style={styles.textStyle}>{text}</Text>
    </View>
  );
};

const Features = () => {
  const featureData = [
    {iconName: 'home-outline', text: 'Bank transfer'},
    {iconName: 'scan-outline', text: 'Scan QR Code'},
    {iconName: 'at-circle-outline', text: 'UPI transfer'},
    {iconName: 'document-text-outline', text: 'View Expenses'},
  ];

  return (
    <View style={{flexDirection: 'row'}}>
      {featureData.map((item, index) => (
        <FeatureItem key={index} iconName={item.iconName} text={item.text} />
      ))}
    </View>
  );
};

export default Features;

const styles = StyleSheet.create({
  featureContainer: {
    justifyContent: 'space-between',
    marginVertical: 24,
    alignItems: 'center',
    flex: 1,
  },
  iconContainer: {
    backgroundColor: 'white',
    borderRadius: 8,
    elevation: 6,
    height: 56,
    aspectRatio: 1,
    // width: '22%',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 8,
  },
  textStyle: {
    flexWrap: 'wrap',
    color: 'gray',
    maxWidth: 56,
    textAlign: 'center',
    fontSize: 14,
  },
});
