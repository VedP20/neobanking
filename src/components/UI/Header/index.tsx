import {StyleSheet, TouchableOpacity, View} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';

const Header = () => {
  return (
    <View style={styles.container}>
      <TouchableOpacity>
        <Icon name="menu-outline" size={28} color="gray" />
      </TouchableOpacity>
      <View style={styles.icons}>
        <TouchableOpacity>
          <Icon name="notifications-outline" size={28} color="gray" />
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name="information-circle-outline" size={28} color="gray" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 56,
    alignItems: 'center',
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingHorizontal: 16,
  },
  icons: {
    flexDirection: 'row',
    gap: 20,
  },
});
