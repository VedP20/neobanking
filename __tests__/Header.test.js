/**
 * @format
 */

import 'react-native';
import React from 'react';
import Header from '../src/components/UI/Header/index';

// Note: import explicitly to use the types shiped with jest.
import { it } from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', async () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
