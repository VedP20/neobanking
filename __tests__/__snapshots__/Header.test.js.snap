// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Ionicons
      color="gray"
      name="menu-outline"
      size={28}
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Ionicons
        color="gray"
        name="notifications-outline"
        size={28}
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Ionicons
        color="gray"
        name="information-circle-outline"
        size={28}
      />
    </View>
  </View>
</View>
`;
